### PR TITLE
fix: arithmetic and equality precedence

### PR DESCRIPTION
**Pull request by brennanjl**
Fixes a bug where equality and assignment operators take a higher precedence than arithmetic operators.

We shouldn't backport this because it will break consensus.

### CODE REVIEW

**Comment by brennanjl** [comment]
converted back to draft as I recognized some other precedence issues

File: parse/grammar/KuneiformParser.g4 [hunk 1]
```diff
@@ -279,34 +279,41 @@ delete_statement:
...
-    | left=sql_expr (PLUS | MINUS) right=sql_expr                                   # arithmetic_sql_expr
-    // setting precedence for logical operations:
+
+    // the rest:
+    | sql_expr NOT? IN LPAREN (sql_expr_list|select_statement) RPAREN               # in_sql_expr
+    | left=sql_expr NOT? (LIKE|ILIKE) right=sql_expr                                # like_sql_expr
+    | element=sql_expr (NOT)? BETWEEN lower=sql_expr AND upper=sql_expr             # between_sql_expr
+    | left=sql_expr (EQUALS | EQUATE | NEQ | LT | LTE | GT | GTE) right=sql_expr    # comparison_sql_expr
+    | left=sql_expr IS NOT? ((DISTINCT FROM right=sql_expr) | NULL | TRUE | FALSE)  # is_sql_expr
+    | <assoc=right> (NOT) sql_expr                                                  # unary_sql_expr
```
Comment (Yaiba) [inline parse/grammar/KuneiformParser.g4:311]: nitpicking, no need for parentheses

File: parse/grammar/KuneiformParser.g4 [hunk 2]
```diff
@@ -279,34 +279,41 @@ delete_statement:
     (WHERE where=sql_expr)?
 ;
 
+// https://docs.kwil.com/docs/kuneiform/operators
 sql_expr:
-    literal type_cast?                                                              # literal_sql_expr
+    // highest precedence:
```
Comment (Yaiba) [inline parse/grammar/KuneiformParser.g4:284]: I suggest we put expr with same shape next to each other, different shape expr has no precedence issues

File: parse/parse_test.go [hunk 3]
```diff
@@ -1952,18 +1973,46 @@ func mustNewDecimal(precision, scale uint16) *types.DataType {
...
+		if isNeg {
+			t *= -1
+		}
+
+		liter := &parse.ExpressionLiteral{
 			Type:  types.IntType,
 			Value: int64(t),
 		}
+
+		if isNeg {
```
Comment (Yaiba) [inline parse/parse_test.go:1989]: 👍

**Review by Yaiba — APPROVED** [review]
lgtm

File: parse/grammar/KuneiformParser.g4 [hunk 4]
```diff
@@ -347,22 +354,29 @@ procedure_block:
...
 ;
 
+// https://docs.kwil.com/docs/kuneiform/operators
 procedure_expr:
-    literal type_cast?                                                                          # literal_procedure_expr
+    // highest precedence:
+    LPAREN procedure_expr RPAREN type_cast?                                                     # paren_procedure_expr
+    | procedure_expr PERIOD IDENTIFIER type_cast?                                               # field_access_procedure_expr
+    | procedure_expr LBRACKET procedure_expr RBRACKET type_cast?                                # array_access_procedure_expr
+    | <assoc=right> (PLUS|MINUS|EXCL) procedure_expr                                            # unary_procedure_expr
```
Comment (Yaiba) [inline parse/grammar/KuneiformParser.g4:363]: is `!323` and `!age` valid in Postgres?

Comment (Yaiba) [reply parse/grammar/KuneiformParser.g4:363]: wait I don't see `!` in the operator table https://www.postgresql.org/docs/16/sql-syntax-lexical.html#SQL-PRECEDENCE-TABLE

Comment (brennanjl) [reply parse/grammar/KuneiformParser.g4:363]: No but we translate it to `not` when we generate code. Also it is only in procedures.

File: parse/grammar/KuneiformParser.g4 (hunk 2 above)
Comment (brennanjl) [reply parse/grammar/KuneiformParser.g4:284]: It's effectively the same thing

File: parse/grammar/KuneiformParser.g4 (hunk 4 above)
Comment (Yaiba) [reply parse/grammar/KuneiformParser.g4:363]: OIC

**Pull request merged**